src/vector.rs

```diff
@@ -426,6 +433,14 @@ impl<S: BaseNum> Vector2<S> {
 }
 
 impl<S: BaseNum> Vector3<S> {
+    #[inline]
+    pub fn to_homogeneous(self) -> Vector4<S> {
```

Comment:
Points and vectors have different representations in homogeneous coordinates (vectors have a `w` coordinate of 0 so they're not affected by translations, points have a `w` coordinate other than 0 (often 1)), so these methods should probably be called `point_to_homogeneous` etc or something similar, or made private.